repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -350,72 +352,69 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         final long startTime = System.currentTimeMillis();
         long streamLimit = containerConfigurationService.getContentBodyReadLimit();
 
-        final MutableHttpServletRequest mutableHttpRequest = MutableHttpServletRequest.wrap((HttpServletRequest) request, streamLimit);
-        final MutableHttpServletResponse mutableHttpResponse = MutableHttpServletResponse.wrap(mutableHttpRequest, (HttpServletResponse) response);
+        final HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request,
+                new BufferedServletInputStream(new LimitedReadInputStream(streamLimit, request.getInputStream())));
+        final HttpServletResponseWrapper wrappedResponse = new HttpServletResponseWrapper((HttpServletResponse) response,
+                ResponseMode.PASSTHROUGH,
```

Comment:
We do call the set methods, and I'm pretty sure this works. However, there isn't much of a reason not to make it mutable, and that gives us a little more power if we need it. So I'll change it.

Also the response is always uncommitted.